modules/openapi-generator/src/test/java/org/openapitools/codegen/java/spring/SpringCodegenTest.java

```diff
@@ -3090,6 +3172,987 @@ public void doCallFluentParentSettersFromChildModel() throws IOException {
+                .toFileAssert()
+                // Setter method assertions
+                .assertMethod("setNames")
+                .hasReturnType("void")
+                .hasParameter("names")
+                .withType("List<String>")
+                .toMethod()
+                .toFileAssert()
+                // Getter method assertions
+                .assertMethod("getNames")
+                .hasReturnType("List<String>")
+                .doesNotHaveParameters()
+                .toFileAssert();
+
+        assertMethod(javaFileAssert, String.class, "alias");
+        assertMethod(javaFileAssert, String.class, "color");
+
+        assertMethod(javaFileAssert, String.class, "stringPattern");
+        assertMethod(javaFileAssert, String.class, "stringMaxMinLength");
+        assertMethod(javaFileAssert, String.class, "stringMinLength");
+        assertMethod(javaFileAssert, String.class, "stringMaxLength");
+        assertMethod(javaFileAssert, String.class, "stringEmail");
+        assertMethod(javaFileAssert, Integer.class, "intMinMax");
+        assertMethod(javaFileAssert, Integer.class, "intMin");
+        assertMethod(javaFileAssert, Integer.class, "intMax");
+        assertMethod(javaFileAssert, BigDecimal.class, "numberMinMax");
+        assertMethod(javaFileAssert, BigDecimal.class, "numberMin");
+        assertMethod(javaFileAssert, BigDecimal.class, "numberMax");
+        assertMethod(javaFileAssert, "Zebra", "zebra");
+
+        assertMethod(javaFileAssert, String.class, "stringPatternNullable");
+        assertMethod(javaFileAssert, String.class, "stringMaxMinLengthNullable");
+        assertMethod(javaFileAssert, String.class, "stringMinLengthNullable");
+        assertMethod(javaFileAssert, String.class, "stringMaxLengthNullable");
+        assertMethod(javaFileAssert, String.class, "stringEmailNullable");
+        assertMethod(javaFileAssert, Integer.class, "intMinMaxNullable");
+        assertMethod(javaFileAssert, Integer.class, "intMinNullable");
+        assertMethod(javaFileAssert, Integer.class, "intMaxNullable");
+        assertMethod(javaFileAssert, BigDecimal.class, "numberMinMaxNullable");
+        assertMethod(javaFileAssert, BigDecimal.class, "numberMinNullable");
+        assertMethod(javaFileAssert, BigDecimal.class, "numberMaxNullable");
+
+    }
+
+    @Test
+    public void testModelsWithNoneOptionalAndNoneOpenApiNullable() throws IOException {
+        File output = Files.createTempDirectory("test").toFile().getCanonicalFile();
+        output.deleteOnExit();
+        String outputPath = output.getAbsolutePath().replace('\\', '/');
+
+        OpenAPI openAPI = new OpenAPIParser()
+                .readLocation("src/test/resources/3_0/issue_14765.yaml", null, new ParseOptions()).getOpenAPI();
+
+        SpringCodegen codegen = new SpringCodegen();
+        codegen.setOutputDir(output.getAbsolutePath());
+        codegen.setOpenApiNullable(false);
+        codegen.setUseOptional(true);
```

Comment:
Is this supposed to be false or am I reading the test name incorrectly?

Reply:
Let me double check